crates/re_log_types/src/path/entity_path_filter.rs

```diff
@@ -369,26 +378,40 @@ impl EntityPathRule {
     #[inline]
     pub fn including_subtree(path: EntityPath) -> Self {
         Self {
+            raw_expression: format!("{path}/**",),
             path,
             include_subtree: true,
         }
     }
 
-    pub fn parse_forgiving(expression: &str) -> Self {
-        let expression = expression.trim();
+    pub fn parse_forgiving(expression: &str, subst_env: &HashMap<String, String>) -> Self {
+        let raw_expression = expression.trim().to_owned();
+
+        // TODO(jleibs): This is a very naive implementation of variable substitution.
+        // unclear if we want to do this here, push this down into `EntityPath::parse`,
+        // or even supported deferred evaluation on the `EntityPath` itself.
+        let mut expression_sub = raw_expression.clone();
+        for (key, value) in subst_env {
+            expression_sub = expression_sub.replace(format!("${key}").as_str(), value);
+            expression_sub = expression_sub.replace(format!("${{{key}}}").as_str(), value);
+        }
```

Comment:
One failure to call out it that `$foo` will also match `$foobar`.

But yeah, solving this in general is a bit tricky, since parsing `${foo}/bar` as an entity path currently results in a warning being printed about unescaped special characters. If we remove that warning though, we could parse, then substitute.

Or we can push the `subst_env` into `EntityPath::parse_forgiving` and handle it there.

I think this is good enough as a first step though, but let's add a TODO linking to an actual issue listing the shortcomings and potential solutions

Reply:
Actually, a better solution occurred to me: if we limit substitutions to the _start_ of each expression, we can make something simple that works well: matching either the entire expression (`$origin`) or stripping the start (`$origin/`).

Motivation: all EntityPath:s are absolute, so it only makes sense to substitute the start of an expression!

If we do that, I think we should change `subst_env: &HashMap<String, String>` to `prefix_subst_env: &HashMap<String, EntityPath>` for clarity